mqtt/mqtt-bridge/src/client.rs

```diff
@@ -165,23 +165,21 @@ where
         let default_topics = self.event_handler.subscriptions();
         self.subscribe(default_topics).await?;
 
-        self.handle_events().await;
+        self.handle_events().await?;
         Ok(())
     }
 
-    async fn handle_events(&mut self) {
+    async fn handle_events(&mut self) -> Result<(), ClientError> {
         debug!("polling bridge client...");
 
-        while let Some(event) = self.client.try_next().await.unwrap_or_else(|e| {
-            // TODO: handle the error by recreating the connection
-            error!(error = %e, "failed to poll events");
-            None
-        }) {
+        while let Some(event) = self.client.try_next().await? {
```

Comment:
I think we had a discussion already and agreed this code will never error out. Could you tell what changed? Did we miss something?